adding troubleshooting steps for pulumi destroy (#4124)
pulumi/pulumi-hugo: themes/default/content/docs/support/troubleshooting.md

This error is often caused by a misconfigured ingress-controller not updating the `status.loadBalancer`
field once the Ingress resource is ready to route traffic.

In some cases, this may be fixed by running `pulumi refresh`.

> #### *Traefik*

Reviewer: ```suggestion
#### Traefik
```

For the Traefik controller, verify that the `kubernetes.ingressEndpoint` config
is [set properly](https://docs.traefik.io/providers/kubernetes-ingress/). This option was
introduced in Traefik 1.7.0.

### Pulumi destroy fails {#pulumi-destroy-fails}

> There are scenarios when `pulumi destroy` will fail to delete resources as expected. This is anticipated due to the nature of cloud provider dependencies, permissions, resources being in a state that prevents their deletion, and when a timeout is not long enough for the cloud provider to complete its operation. Review the output to identify which resources were not deleted and consider the following steps depending on the nature of the failure.

Reviewer: ```suggestion
There are scenarios when `pulumi destroy` will fail to delete resources as expected. This is anticipated due to the nature of cloud provider dependencies, permissions, resources being in a state that prevents their deletion, or when a timeout is not long enough for the cloud provider to complete its operation. Review the output to identify which resources were not deleted and consider the following steps depending on the nature of the failure.
```

#### Check to see if a resource was deleted after all

Some resources take time to be removed. Common examples include CloudFront Lambda@Edge functions, which will fail to `destroy` but will eventually disappear without requiring further action. In these cases, you can wait and run `pulumi refresh` to see if the cloud provider was able to remove the resource.

#### Check dependencies

If the issue is due to dependencies, identify and delete the dependent external resources manually. This may involve navigating the cloud provider's console or using its CLI to pinpoint and resolve these dependencies.

#### Empty or adjust resources

> If a resource cannot be deleted because it contains data, or there are network interfaces or other dependencies not automatically removed by Pulumi. Common examples include deleting VPCs with EINs attached elsewhere and deleting a security group when it is in use. You will need to evaluate the dependencies given your failure and take the necessary actions to resolve this on each provider resource.

Reviewer: ```suggestion
Occasionally a resource cannot be deleted because it contains data or uses network interfaces or other dependencies managed outside the stack. Common examples include deleting VPCs with EINs attached elsewhere or deleting a security group when it is in use. You will need to evaluate the dependencies given the failure and take the necessary actions to resolve this on each provider resource.
```

#### Delete resources manually

> For each resource that couldn't be deleted, use the cloud provider's console or CLI to manually delete it. This might be necessary for resources in a locked state or those with specific permissions preventing automated deletion.

Reviewer: ```suggestion
For each resource that couldn't be deleted, use the cloud provider's console or CLI to manually delete it. This may be necessary for resources in a locked state or those with specific permissions preventing automated deletion.
```

This warning may be benign. However, if you are experiencing crashes or hangs in Pulumi (especially in Node.js version 12.11.0 and above) and you see this warning, then it is likely that the older version of Pulumi is the issue.

A warning is issued so as to not break existing code that is functioning properly. However, the root cause of this problem

> pertains to undefined behavior in the Node.js runtime, so apparently working code today may begin crashing or hanging in the future. We recommend updating your code to ensure your Pulumi program works reliably.

Reviewer: ```suggestion
pertains to undefined behavior in the Node.js runtime, so code that works today may begin crashing or hanging in the future. We recommend updating your code to ensure your Pulumi program works reliably.
```
